packages/central-server/src/apiV2/GETHandler/helpers.js

```diff
@@ -86,19 +86,51 @@ const getForeignKeyColumnName = foreignTable => {
   return exceptions[foreignTable] || `${foreignTable}_id`;
 };
 
+const addJoin = (
+  recordType,
+  recordTypesInQuery,
+  multiJoin,
+  baseRecordType,
+  customJoinConditions,
+  joinType,
+) => {
+  if (!recordTypesInQuery.has(recordType)) {
+    const join = customJoinConditions[recordType];
+    const joinCondition = join.through
+      ? [join.foreignTable, join.foreignKey]
+      : [`${recordType}.id`, `${baseRecordType}.${getForeignKeyColumnName(recordType)}`];
```

Comment:
This is going to ignore the old format `customJoinConditions`

If `join.through` doesn't exist currently we generate a default `joinCondition`, but for the old format, we would want `joinCondition = join`